components/livekit/src/component.ts

```diff
@@ -0,0 +1,330 @@
+import { ILoggerComponent } from '@well-known-components/interfaces'
+import {
+  AccessToken,
+  IngressClient,
+  IngressInfo,
+  IngressInput,
+  ParticipantInfo,
+  Room,
+  RoomServiceClient,
+  WebhookReceiver
+} from 'livekit-server-sdk'
+import { isErrorWithMessage } from '@dcl/core-commons'
+
+import {
+  CreateIngressOptions,
+  CreateRoomOptions,
+  GenerateCredentialsOptions,
+  ILivekitComponent,
+  LivekitComponentOptions,
+  LivekitCredentials,
+  UpdateParticipantOptions
+} from './types'
+import { LivekitIngressNotFoundError, LivekitWebhookVerificationError } from './errors'
+
+/**
+ * Creates a LiveKit component for managing rooms, participants, and credentials
+ *
+ * @example
+ * ```typescript
+ * const livekit = await createLivekitComponent(
+ *   { logs },
+ *   {
+ *     settings: {
+ *       host: 'wss://livekit.example.com',
+ *       apiKey: 'your-api-key',
+ *       secret: 'your-api-secret'
+ *     }
+ *   }
+ * )
+ *
+ * // Generate credentials for a participant
+ * const credentials = await livekit.generateCredentials({
+ *   identity: 'user-123',
+ *   roomName: 'my-room',
+ *   permissions: { canPublish: true, canSubscribe: true }
+ * })
+ * ```
+ */
+export async function createLivekitComponent(
+  components: { logs: ILoggerComponent },
+  options: LivekitComponentOptions
+): Promise<ILivekitComponent> {
+  const { logs } = components
+  const { settings } = options
+
+  const logger = logs.getLogger('livekit-component')
+
+  // Normalize host URL to ensure it starts with wss://
+  const normalizedHost = settings.host.startsWith('wss://') ? settings.host : `wss://${settings.host}`
+
+  // Initialize LiveKit clients
+  const roomClient = new RoomServiceClient(normalizedHost, settings.apiKey, settings.secret)
+  const ingressClient = new IngressClient(normalizedHost, settings.apiKey, settings.secret)
+  const webhookReceiver = new WebhookReceiver(settings.apiKey, settings.secret)
+
+  // ============= CREDENTIALS =============
+
+  async function generateCredentials(opts: GenerateCredentialsOptions): Promise<LivekitCredentials> {
```

Comment:
As the `identity` and `roomName` and required, and not options, what do you think about changing the function signature to be:

```suggestion
  async function generateCredentials(identity: string, roomName: string, opts: GenerateCredentialsOptions): Promise<LivekitCredentials> {
```